### PR TITLE
jwt-auth: add an interface to create a public key object 

### PR DESCRIPTION
**Pull request by mtakigiku**
added `Pubkeys::CreateFrom()`

### CODE REVIEW

File: src/envoy/auth/jwt.cc [hunk 1]
```diff
@@ -433,6 +433,18 @@ std::unique_ptr<Pubkeys> Pubkeys::CreateFromJwks(const std::string &pkey_jwks) {
...
 
+std::unique_ptr<Pubkeys> Pubkeys::CreateFrom(const std::string &pkey,
+                                             Type type) {
+  switch (type) {
+    case Type::JWKS:
+      return CreateFromJwks(pkey);
+    case Type::PEM:
+      return CreateFromPem(pkey);
+    default:
+      abort();
```
Comment (lizan) [inline src/envoy/auth/jwt.cc]: use PANIC macro and indicate the reason.

File: src/envoy/auth/jwt.h [hunk 2]
```diff
@@ -205,9 +205,14 @@ class JwtVerifier : public WithStatus {
 //   if(keys->GetStatus() == Status::OK) { ... }
 class Pubkeys : public WithStatus {
  public:
+  // Format of public key.
+  enum Type { PEM, JWKS };
+
   Pubkeys(){};
   static std::unique_ptr<Pubkeys> CreateFromPem(const std::string& pkey_pem);
```
Comment (lizan) [inline src/envoy/auth/jwt.h]: make these two private?

Comment (mtakigiku) [reply src/envoy/auth/jwt.h]: done. Removed these two rather than making them private.

File: src/envoy/auth/jwt.cc (hunk 1 above)
Comment (mtakigiku) [reply src/envoy/auth/jwt.cc]: done

**Pull request merged**